classes/profile.php

```diff
@@ -80,6 +83,21 @@ protected function get_flamedatad3(): string {
         return json_decode($this->get_flamedatad3json());
     }
 
+    /**
+     * Set the parameters field. Will be truncated if longer than MAX_STR_LEN bytes.
+     *
+     * @param string $parameters
+     * @throws \coding_exception
+     */
+    protected function set_parameters(string $parameters) {
+        /* The database field can only store MAX_STR_LEN bytes. So if the string is longer, we chop it off and add
```

Comment:
> store MAX_STR_LEN bytes

I believe it has to do with the number of characters stored, as opposed to the bytes used for storage, at least from the https://docs.moodle.org/dev/XMLDB_column_types and the actual checks being performed (e.g. I don't think we are checking the size of bytes here)

Reply:
Changed DB field to TEXT